src/m365/pp/commands/website/website-webrole-remove.ts

```diff
@@ -0,0 +1,192 @@
+  websiteId?: string;
+  websiteName?: string;
+  asAdmin?: boolean;
+  force?: boolean;
+}
+
+class PpWebSiteWebRoleRemoveCommand extends PowerPlatformCommand {
+
+  public get name(): string {
+    return commands.WEBSITE_WEBROLE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the specified webrole';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        id: typeof args.options.id !== 'undefined',
+        websiteId: typeof args.options.websiteId !== 'undefined',
+        websiteName: typeof args.options.name !== 'undefined',
+        asAdmin: !!args.options.asAdmin,
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environmentName <environmentName>'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '--websiteId [websiteId]'
+      },
+      {
+        option: '--websiteName [websiteName]'
+      },
+      {
+        option: '--asAdmin'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['websiteId', 'websiteName'] }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+        if (args.options.websiteId && !validation.isValidGuid(args.options.websiteId)) {
+          return `${args.options.websiteId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: any): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Removing websrole '${args.options.id}'...`);
+    }
+
+    if (args.options.force) {
+      await this.deleteWebSiteWebRole(args);
+    }
+    else {
+      const result = await cli.promptForConfirmation({ message: `Are you sure you want to remove webrole '${args.options.id}'?` });
+
+      if (result) {
+        await this.deleteWebSiteWebRole(args);
+      }
+    }
+  }
+
+  private async getWebSiteId(dynamicsApiUrl: string, args: CommandArgs): Promise<any> {
+    if (args.options.websiteId) {
+      return args.options.websiteId;
+    }
+
+    const options: PpWebSiteOptions = {
+      environmentName: args.options.environmentName,
+      id: args.options.websiteId,
+      name: args.options.websiteName,
+      output: 'json'
+    };
+
+    const requestOptions: CliRequestOptions = {
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    if (options.name) {
```

Comment:
Let's remove this nullcheck as we know that args.options.websiteName should be filled here, thanks to the validator. Let's use `args.options.websiteName!` if necessary...